test/index.js

```diff
@@ -0,0 +1,244 @@
+const assert = require('assert');
+const helpers = require('../helpers');
+
+describe('Apostrophe-a3-schema-exporter', function() {
+  it('should handle arrays', () => {
+    const field = {
+      name: 'content-links',
+      type: 'array',
+      schema: [
+        {
+          name: 'url',
+          label: 'Url',
+          type: 'url'
+        },
+        {
+          name: 'description',
+          label: 'Description',
+          type: 'string'
+        }
+      ],
+      label: 'Content'
+    };
+    const generatedArray = helpers.handleArray(field);
+
+    assert.deepEqual(generatedArray, {
+      type: 'array',
+      label: 'Content',
+      fields: {
+        add: {
+          url: {
+            label: 'Url',
+            type: 'url'
+          },
+          description: {
+            label: 'Description',
+            type: 'string'
+          }
+        }
+      }
+    });
+  });
+
+  it('should handle tags as an array', () => {
+    const field = {
+      type: 'tags',
+      name: 'tags',
+      label: 'Tags'
+    };
+    const generatedArray = helpers.handleArray(field);
+
+    assert.deepEqual(generatedArray, {
+      type: 'array',
+      label: 'Tags',
+      fields: {
+        add: {
+          tag: {
+            type: 'string',
+            label: 'Tag'
+          }
+        }
+      }
+    });
+  });
+
+  it('should handle areas', async () => {
+    const props = {
+      label: 'Widgets',
+      type: 'area',
+      options: {
+        widgets: {
+          'apostrophe-rich-text': { toolbar: [ 'Undo', 'Redo' ] },
+          'apostrophe-images': { size: 'full' },
+          'apostrophe-video': {}
+        }
+      }
+    };
+    const generatedArea = helpers.handleArea(props);
+
+    assert.deepEqual(generatedArea, {
+      label: 'Widgets',
+      type: 'area',
+      options: {
+        widgets: {
+          '@apostrophecms/rich-text': { toolbar: [ 'Undo', 'Redo' ] },
+          '@apostrophecms/image': { size: 'full' },
+          '@apostrophecms/video': {}
+        }
+      }
+    });
+  });
+
+  it('should handle joinByOne', async () => {
+    const cur = {
+      type: 'joinByOne',
+      name: '_article',
+      label: 'Article',
+      idField: 'articleId'
+    };
+    const generatedRelationship = helpers.handleRelationship(cur);
+
+    assert(generatedRelationship, {
+      max: 1,
+      type: 'relationship'
+    });
```

Comment:
what about the `withType` and the `label`attributes?

Reply:
the goal of this test is to check joinByOne is converted to max
withType is tested in another test